esphome/components/rtsp_server/__init__.py

```diff
@@ -0,0 +1,36 @@
+import esphome.config_validation as cv
+import esphome.codegen as cg
+from esphome.const import (
+    CONF_ID,
+    CONF_PORT
+);
+from esphome.core import coroutine_with_priority, CORE
+from esphome.components import esp32_camera
+
+
+CODEOWNERS = ["@crossan007"]
+DEPENDENCIES = ["network"]
+AUTO_LOAD = ["async_tcp"]
+
+rtsp_server_ns = cg.esphome_ns.namespace("rtsp_server")
+RTSPServer = rtsp_server_ns.class_("RTSPServer", cg.Component)
+
+CONF_RTSP_SERVER_ID = "rtsp_server_id"
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(): cv.declare_id(RTSPServer),
+        cv.Optional(CONF_PORT, default=8555): cv.port,
+        cv.Required("camera"): cv.use_id(esp32_camera.ESP32Camera),
```

Comment:
1. Config keys should be in constants, like `CONF_CAMERA` (for example below `AUTO_LOAD` line).
2. There's no need for `Required`, esphome can auto-wire the camera if there's only one in the config (which is likely the case for all configs). 
```suggestion
        cv.GenerateID(CONF_CAMERA): cv.use_id(esp32_camera.ESP32Camera),
```